[DynamoDB] 2.1 updates (#7102)
Particular/docs.particular.net: persistence/dynamodb/sagas_consistency_dynamodb_[2.1,).partial.md

> Amazon DynamoDB supports two read models: strongly consistent and eventually consistent. By default, the AWS SDK uses eventual consistency, which returns the most recently committed version of a record—but without guaranteeing that the latest write is visible at the time of the read. While this model is sufficient for many applications, it can introduce correctness issues when used with concurrency control mechanisms like optimistic locking.

Reviewer: ```suggestion
Amazon DynamoDB supports two read models: strongly consistent and eventually consistent. By default, the AWS SDK uses eventual consistency, which returns the most recently committed version of a record but does not guarantee that the latest write is visible at the time of the read. While this model is sufficient for many applications, it can introduce correctness issues when used in conjunction with concurrency control mechanisms, such as optimistic locking.
```
